sql/threadpool_common.cc

```diff
@@ -191,6 +192,7 @@ void threadpool_remove_connection(THD *thd) {
 
 #ifdef HAVE_PSI_THREAD_INTERFACE
   PSI_THREAD_CALL(delete_thread)(thd->get_psi());
+  mysql_thread_set_psi_THD(nullptr);
```

Comment:
Likewise, please check if this needs to be outside of the `#ifdef HAVE_PSI_THREAD_INTERFACE`